File: AppExtension/WidgetProvider.cpp
#include "pch.h"
#include "WidgetProvider.h"
#if __has_include("WidgetProvider.g.cpp")
#include "WidgetProvider.g.cpp"
#endif

namespace winrt::PackagedOOPWinRTServerAppExtension::implementation
{
    WidgetProvider::WidgetProvider()
    {
        auto provider = get_strong().try_as<winrt::SDK::IWidgetProvider>();
        if (provider == nullptr)
        {

        }
    }
    winrt::hstring WidgetProvider::SayHello()
    {
        return L"Greeting from C++ extension!";
    }
}

File: AppExtensionCSharp/WidgetProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AppExtensionCSharp
{
    public sealed class WidgetProvider : SDK.IWidgetProvider
    {
        public WidgetProvider() { }

        public string SayHello()
        {
            return "Greeting from C# Extension!";
        }
    }
}
